test/test_core.py

```diff
@@ -88,6 +88,31 @@ def decorated(self):
   return decorated
 
 
+def with_both_longjmp(f):
```

Comment:
`with_both_eh_sjlj` enables longjmp. That's how we have been testing all longjmps (with wasm32) so far. This has an unnecessary side effect of enabling EH in longjmp tests as well, but that hasn't been an issue. And your `with_both_longjmp` also seems to enable `-fwasm-exceptions`..? Currently the only difference between `with_both_longjmp` and `with_both_eh_sjlj` is that `with_both_longjmp` allows wasm64.

If we really don't want to enable exceptions on longjmp tests, we can maybe split decorators into `with_both_eh` and `with_both_sjlj` (or `with_both_longjmp`). But I'm not sure if that's necessary... If we only want to disable wasm64 for EH I guess we can split the decorators temporarily? But we can eventually merge it when wasm64 supports EH.

Reply:
The reason I need this is because if we enable EH then I can't run the longjmp tests under wasm64.   `with_both_longjmp` is a narrower requirement that makes sense for tests that just use longjmp. 

With wasm64 we don't support emscripten EH, unlike wasm32 which does.

I can call it  `with_both_sjlj` if you prefer?    But the setting it controls is called SUPPORT_LONGJMP so so it made sense for me to call it that.

Reply:
> Currently the only difference between `with_both_longjmp` and `with_both_eh_sjlj` is that `with_both_longjmp` allows wasm64.

The other difference (and the important one here) is that `with_both_longjmp` doesn't set `self.set_setting('DISABLE_EXCEPTION_CATCHING', 0)`

Reply:
I renamed the new function to `with_both_sjlj` and added a comment that should make clear why its needed in addition to `with_both_eh_sjlj`.

Reply:
Then will we remove this after Wasm64 supports Emscripten EH? Or you are planning to test longjmp tests with `with_both_sjlj`? If so, do we need `sjlj` part in `with_both_eh_sjlj`, given that it is not used to test longjmps at all? (That's what I meant by splitting the decorators into `with_both_eh` and `with_both_sjlj`. I don't mind whether it is `sjlj` or `longjmp`)

Reply:
Ah, so you are saying that with_both_eh_sjlj is maybe used on tests that don't need longjmp at all?  I'll take a look, and see if I can rename that to just `with_both_eh` then.

Reply:
There are at least 3 tests with the `with_both_eh_sjlj` marker that do need longjmp support: `test_exceptions_longjmp1_wasm`, `test_exceptions_longjmp2_wasm`, `test_exceptions_longjmp4_wasm`... they fail if both are not set to the same backend,  so maybe best to leave that as enabling both EH and SJLJ?

Reply:
Once Wasm64 supports Emscripten EH (it may never do) then we can remove this second decorator, yes

Reply:
> There are at least 3 tests with the `with_both_eh_sjlj` marker that do need longjmp support: `test_exceptions_longjmp1_wasm`, `test_exceptions_longjmp2_wasm`, `test_exceptions_longjmp4_wasm`... they fail if both are not set to the same backend, so maybe best to leave that as enabling both EH and SJLJ?

Ah right I had forgotten about them... Yeah then at the moment we can't help having `with_both_eh_sjlj` and `with_both_sjlj`.